NGF: Add document for upstream http2 (#2021)
nginx/documentation: content/ngf/reference/permissions.md

- **EmptyDir**: Read-write (required for NGINX operation)
- **Secret/ConfigMap/Projected**: Read-only

{{< call-out "note" >}}

> Kubernetes is generally not recommended for hostile multi-tenant environments and NGINX is designed to treat upstreams as trusted. If you need a dataplane that doesn't treat upstreams as trusted, you may want to explore alternative solutions.

Reviewer: ```suggestion
We don't recommend Kubernetes for hostile multi-tenant environments, and NGINX is designed to treat upstreams as trusted. If you need a data plane that doesn't treat upstreams as trusted, you may want to explore alternative solutions.
```

If we keep this statement, make it less vague. And "data plane" two words as per our style guide.

Author: I'm going to remove this statement. Feels like there isn't a strong push for wanting it from any maintainers besides a voice from a community member, and overall this statement feels unnecessary and out of place.

> {{< /call-out >}}

Author: added this note here for the request by the user @salonichf5, do you have any thoughts? Should this even go in our docs here or maybe should it just live in our github docs somewhere? I don't really feel the need of highlighting this statement, and am on the border of if its necessary at all and might do more harm than good.

Reviewer 2: can we just remove this part "If you need a dataplane that doesn't treat upstreams as trusted, you may want to explore alternative solutions." 

we shouldn't be recommending that I feel. The short callout is fine by me.

Author: Ditto from above: I'm going to remove this statement. Feels like there isn't a strong push for wanting it from any maintainers besides a voice from a community member, and overall this statement feels unnecessary and out of place.
